src/Inline.cpp

```diff
@@ -14,83 +14,80 @@ using std::set;
 using std::string;
 using std::vector;
 
-class Inliner : public IRMutator {
-    using IRMutator::visit;
-
-    Function func;
-
-    // Sanity check that this is a reasonable function to inline
-    void check(Function f) {
-
-        internal_assert(f.can_be_inlined()) << "Illegal to inline " << f.name() << "\n";
-
-        const Schedule &s = f.schedule();
+// Sanity check that this is a reasonable function to inline
+void validate_schedule_inlined_function(Function f) {
```

Comment:
I just lifted the inline check into a separate function, so that I can use it inside src/ScheduleFunctions.cpp